crates/engine/tree/src/tree/root.rs

```diff
@@ -524,6 +530,16 @@ where
                             return Ok((root, trie_updates));
                         }
                     }
+                    StateRootMessage::ProofCalculationError(e) => {
+                        return Err(ParallelStateRootError::Other(format!(
+                            "could not calculate multiproof: {e:?}"
+                        )))
+                    }
+                    StateRootMessage::RootCalculationError(e) => {
+                        return Err(ParallelStateRootError::Other(format!(
```

Comment:
think we should introduce a custom error enum for the state root task, we seem to abuse the `ParallelStateRootError::Other`

Reply:
@shekhirin, this is a good idea. Can we create this custom error enum within the state root task?